person.rb

```diff
@@ -0,0 +1,35 @@
+require './nameable'
+
+# Parent class for teacher and student
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :age, :name
+
+  def initialize(age, name = 'Unknown', parent_permision: true)
+    super()
+    @id = Random.rand(1..100)
+    @name = name
+    @age = age
+    @rentals = []
+    _parent_permision = parent_permision
+  end
+
+  def can_use_services?
+    of_age? || parent_permision
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [ ] Nice attempt trying to implement the many-to-many relationship between the `Book` class and the `Person` class through the `Rental` class. However, for this relationship to be properly implemented, the `Book` and `Person` classes need to have their `add_rental` method enable them to create a new `Rental` object instead of pushing the rental object to the `rentals` array. 👍 Kindly change what this method does. You can read [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to have a better understanding. 👍